src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Vector;
+
+import duke.exception.SaveException;
+import duke.task.Task;
+
+public class TaskList extends Vector<Task> {
+    public static final long serialVersionUID = 1L;
+    protected Storage storage;
+
+    public TaskList(Storage storage) {
+        super();
+        setStorage(storage);
+    }
+
+    public void setStorage(Storage storage) {
+        this.storage = storage;
+    }
+
+    /*
+     * Save file on all the modification actions
+     */
```

Comment:
You might want to move this to a Javadoc comment for the class itself

Reply:
Thanks for the advice. This has been fixed in the latest commit.